packages/apollo-gateway/src/__tests__/integration/logger.test.ts

```diff
@@ -0,0 +1,125 @@
+import { ApolloGateway } from '../..';
+import { Logger } from "apollo-server-types";
+import { PassThrough } from "stream";
+
+import * as winston from "winston";
+import WinstonTransport from 'winston-transport';
+import * as bunyan from "bunyan";
+import * as loglevel from "loglevel";
+// We are testing an older version of `log4js` which uses older ECMAScript
+// in order to still support testing on Node.js 6.
+// This should be updated when bump the semver major for AS3.
+import * as log4js from "log4js";
+
+const LOWEST_LOG_LEVEL = "debug";
+
+const KNOWN_DEBUG_MESSAGE = "Checking service definitions...";
+
+async function triggerKnownDebugMessage(logger: Logger) {
+  // Trigger a known error.
+  // This is a bit britle since it merely leverages a known debug log
+  // message outside of the constructor, but it seemed worth testing
+  // the compatibility with `ApolloGateway` itself rather than genericly.
+  // The error does not matter, so it is caught and ignored.
```

Comment:
```suggestion
  // This is a bit brittle since it merely leverages a known debug log
  // message outside of the constructor, but it seemed worth testing
  // the compatibility with `ApolloGateway` itself rather than generically.
  // The error does not matter, so it is caught and ignored.
```